### DISCUSSION TITLE
problem with Xorg/lighdm with Xubuntu 20.04

### DISCUSSION THREAD

**Discussion by FranckGoirand** [ltsp/ltsp#330] [General]
Good morning

**_The installation is as follows:_**

Tests with images on Xubuntu 20.04 and lightdm and on Ubuntu 20.04 and gdm. The images were made from VM.
The Server is on Ubuntu 20.04 server.

**The problem is the following:** 
During the first boot under Xubuntu the computer boots perfectly and I can login in lightdm and have the graphical interface XFCE, everything works perfectly. I can use this computer for hours without any problem.
The problem occurs when I reboot a second time, the computer tries to launch X but crashes. If I try several times to reboot this computer or another one with the Xubuntu 20.04 image it doesn't work.

**Here are the two solutions I found to make it work:**
1. If I start this same computer with the Ubuntu 20.04 image and GDM everything works perfectly well. I can reboot as many times as I want it works each time.  
2. If I reboot the Ubuntu 20.04 server then I can run the Xubuntu image but only once. To make the Xubuntu image work again I have to reboot the server !!!!.

Considering that computers have small processors (Atom 1900) I want to run on Xubuntu.

Here are some log files :

**seat0-greeter.log**
```
** Message: 17:27:47.232: Starting lightdm-gtk-greeter 2.0.6 (Sep 18 2018, 01:17:10)
** Message: 17:27:47.247: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/01_ubuntu.conf
** Message: 17:27:47.247: [Configuration] Reading file: /usr/share/lightdm/lightdm-gtk-greeter.conf.d/30_xubuntu.conf
** Message: 17:27:47.248: [Configuration] Reading file: /etc/lightdm/lightdm-gtk-greeter.conf
** (lightdm-gtk-greeter:952): WARNING **: 17:27:47.991: [PIDs] Failed to execute command: /usr/lib/at-spi2-core/at-spi-bus-launcher
** (lightdm-gtk-greeter:952): WARNING **: 17:27:47.996: [PIDs] Failed to execute command: upstart
Gdk-Message: 17:29:35.330: lightdm-gtk-greeter: Fatal IO error 0 (Succès) on X server :0.

```

**Xorg.0.log**

```
[    19.929] (**) modeset(G0): DPI set to (83, 84)
[    19.929] (II) Loading sub module "fb"
[    19.929] (II) LoadModule: "fb"
[    19.929] (II) Loading /usr/lib/xorg/modules/libfb.so
[    19.930] (II) Module fb: vendor="X.Org Foundation"
[    19.930] 	compiled for 1.20.8, module version = 1.0.0
[    19.930] 	ABI class: X.Org ANSI C Emulation, version 0.4
[    19.930] (EE) 
[    19.930] (EE) Backtrace:
[    19.931] (EE) 0: /usr/lib/xorg/Xorg (OsLookupColor+0x13c) [0x561eda5ce11c]
[    19.933] (EE) 1: /lib/x86_64-linux-gnu/libpthread.so.0 (funlockfile+0x60) [0x7ff7dacb341f]
[    19.935] (EE) 2: /lib/x86_64-linux-gnu/libc.so.6 (gsignal+0xcb) [0x7ff7daaf018b]
[    19.936] (EE) 3: /lib/x86_64-linux-gnu/libc.so.6 (abort+0x12b) [0x7ff7daacf859]
[    19.937] (EE) unw_get_proc_name failed: no unwind info found [-10]
[    19.937] (EE) 4: /lib/x86_64-linux-gnu/libc.so.6 (?+0x0) [0x7ff7daacf71a]
[    19.938] (EE) 5: /lib/x86_64-linux-gnu/libc.so.6 (__assert_fail+0x46) [0x7ff7daae0f36]
[    19.939] (EE) 6: /usr/lib/xorg/Xorg (dixRegisterPrivateKey+0x239) [0x561eda48ba79]
[    19.939] (EE) 7: /usr/lib/xorg/modules/libglamoregl.so (glamor_init+0xcf) [0x7ff7c84b035f]
[    19.940] (EE) unw_get_proc_name failed: no unwind info found [-10]
[    19.940] (EE) 8: /usr/lib/xorg/modules/drivers/modesetting_drv.so (?+0x0) [0x7ff7da492740]
[    19.941] (EE) unw_get_proc_name failed: no unwind info found [-10]
[    19.941] (EE) 9: /usr/lib/xorg/modules/drivers/modesetting_drv.so (?+0x0) [0x7ff7da48ab80]
[    19.941] (EE) 10: /usr/lib/xorg/Xorg (AddGPUScreen+0xf5) [0x561eda46d2d5]
[    19.942] (EE) 11: /usr/lib/xorg/Xorg (xf86PlatformMatchDriver+0xa44) [0x561eda4c8ca4]
[    19.942] (EE) 12: /usr/lib/xorg/Xorg (xf86PlatformDeviceCheckBusID+0x225) [0x561eda4cdfe5]
[    19.942] (EE) 13: /usr/lib/xorg/Xorg (config_fini+0xa4a) [0x561eda4ca88a]
[    19.943] (EE) 14: /usr/lib/xorg/Xorg (config_fini+0x1408) [0x561eda4cbcf8]
[    19.943] (EE) 15: /usr/lib/xorg/Xorg (OsCleanup+0x5c1) [0x561eda5cefa1]
[    19.943] (EE) 16: /usr/lib/xorg/Xorg (WaitForSomething+0x193) [0x561eda5c7803]
[    19.943] (EE) 17: /usr/lib/xorg/Xorg (SendErrorToClient+0x117) [0x561eda46ccf7]
[    19.944] (EE) 18: /usr/lib/xorg/Xorg (InitFonts+0x3b4) [0x561eda470fc4]
[    19.945] (EE) 19: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xf3) [0x7ff7daad10b3]
[    19.945] (EE) 20: /usr/lib/xorg/Xorg (_start+0x2e) [0x561eda45aa2e]
[    19.945] (EE) 
[    19.945] (EE) 
Fatal server error:
[    19.945] (EE) Caught signal 6 (Aborted). Server aborting
[    19.946] (EE) 
[    19.946] (EE) 
Please consult the The X.Org Foundation support 
	 at http://wiki.x.org
 for help. 
[    19.946] (EE) Please also check the log file at "/var/log/Xorg.0.log" for additional information.
[    19.946] (EE) 
[    40.557] (EE) Server terminated with error (1). Closing log 
```

and ltsp.conf :

```
[common]

[40:62:31:13:3c:a9]   <--- this is my computer where i use
IMAGE_TO_RAM=0
DEBUG_LOG=1
DEBUG_SHELL=1
DEFAULT_IMAGE="xubuntu-boni.vmdk"
MENU_TIMEOUT="2000"

```

Thank you in advance for your help

Best regards.

**Comment by alkisg** [comment]
Hi, this doesn't sound LTSP related. Some questions:

- Does it happen with a different PC that has a different graphics card? If not, that would indicate a driver issue. You could even try with a VM, e.g. a VirtualBox virtual client.
- Do you have autologin enabled? If yes, try without it, as the user saves state on the server (/home/username), while the client (/) doesn't. Also, some window managers like xfwm and marco have bugs but are only started after login
- Are your images clean installations, or upgrades from 18.04? Why do you have upstart listed there in the errors, it shouldn't be used anymore in recent Ubuntu versions...

In Ubuntu MATE 18.04 there was a recent bug that I reported in https://bugs.launchpad.net/ubuntu/+source/slick-greeter/+bug/1902879. It got solved within a week. I don't know if lightdm-gtk-greeter has a similar issue or not.

In general, Ubuntu MATE is as light as Xubuntu, but more people test it, so you might want to use that one if Xubuntu has issues (unrelated to LTSP)...

**Reply by FranckGoirand** [reply]
Hi alkisg

Sorry for the delay in answering your questions. I am glad that Lichtmacher confirms my problem.

To answer your question quickly:
- When I use a client in VM in VMWARE under the same conditions everything works perfectly well.
- I don't have autlogin enabled
- My image is a clean installation since Xubuntu 20.04

What I observed in addition :
- 30 minutes after booting up the computer, the lightdm login screen appears normally

I will test your solution today and get back to you shortly.

**Comment by Lichtmacher** [comment]
I've exactly the same problem.

If lightdm has the same problem as slick-greeter, which seems to be so:
https://bugs.launchpad.net/ubuntu/+source/slick-greeter/+bug/1902879/comments/1

Is it configurable or is it possible to change to a different Loginmanager?

In
https://bugs.launchpad.net/ubuntu/+source/systemd/+bug/1830746/comments/17
you mention a limits.conf.
Is that /etc/security/limits.conf?

I tried settings like
*                soft    memlock         262144
*                hard    memlock         262144
but it didn't help.

Best regards

--------

Clean Install of xubuntu-20.04.1-desktop-amd64.iso in an virtualbox machine.

root@LTS:~# add-apt-repository ppa:ltsp
root@LTS:~# apt update
root@LTS:~# apt install --install-recommends ltsp ltsp-binaries dnsmasq nfs-kernel-server openssh-server squashfs-tools ethtool net-tools epoptes
root@LTS:~# ltsp dnsmasq
root@LTS:~# ltsp image /
root@LTS:~# ltsp ipxe
root@LTS:~# ltsp nfs
root@LTS:~# ltsp initrd

One boot is possible. The second ends in a black screen, some times blinking ASCII. Mouse and keyboard disfunctional. (One time it booted a second time). Booting with HDMI-connector disconnect you can have luck of a further possible login: plug it in after boot, may be lightdm starts.

I have different Clients with the same problem:
NUC 5i3RYH with Intel HD Graphics 5500
NUC7i3DNKE with Intel HD Graphics 620
Shuttle XS35GTA V2 with Atom D2700 and AMD Radeon HD 7410M

Funny thing: Atom D525-Mainboards (Gigabyte GA-D525TUD) work like a charm, but they are slow.


After booting a NUC with disconnected HDMI I got that in /var/log/apport.log:

ERROR: apport (pid 1120) Tue Nov 24 13:24:26 2020: executable: /usr/lib/xorg/Xorg (command line "/usr/lib/xorg/Xorg -core :0 -seat seat0 -auth /var/run/lightdm/root/:0 -nolisten tcp vt7 -novtswitch")
ERROR: apport (pid 1120) Tue Nov 24 13:24:26 2020: is_closing_session(): no DBUS_SESSION_BUS_ADDRESS in environment

**Comment by alkisg** [comment]
LightDM is the display manager. It is the "backend". It doesn't display anything on screen. It does NOT have an issue.
Slick-greeter (Ubuntu MATE) and lightdm-gtk-greeter (Xubuntu) are both front-ends of LightDM. The front-ends are called "greeters" and are the ones that display the login screen. They both had issues, and I think they're both fixed.

If you're using chrootless, try to fully update your server with `sudo apt update; sudo apt full-upgrade` and then run `ltsp image /`.
You can easily install slick-greeter and set it as the default over lightdm-gtk-greeter if you wish; google for "lightdm default greeter", it's not specific to LTSP.

**Reply by FranckGoirand** [reply]
Here is my feedback after installing slick-greeter

- On the VM it works perfectly well, I have the slick-greeter GUI and everything works well.
- On the workstation, always the same problem, I can see the mouse pointer for a few moments, then black screen AND ONLY after arround 30 minutes i can see the screen login.

Here are the logs:

```
[+1,26s] DEBUG: slick-greeter.vala:705: Creating Slick Greeter
[+1,26s] DEBUG: slick-greeter.vala:55: Creating background surface
Gdk-Message: 14:29:02.031: slick-greeter: Fatal IO error 11 (Ressource temporairement non disponible) on X server :0.
```
Xorg.0.log
```(==) Using system config directory "/usr/share/X11/xorg.conf.d"
Xorg: ../../../../dix/privates.c:384: dixRegisterPrivateKey: Assertion `!global_keys[type].created' failed.



[    19.749] (EE) Backtrace:
[    19.750] (EE) 0: /usr/lib/xorg/Xorg (OsLookupColor+0x13c) [0x560c43ab211c]
[    19.753] (EE) 1: /lib/x86_64-linux-gnu/libpthread.so.0 (funlockfile+0x60) [0x7f195c4be41f]
[    19.754] (EE) 2: /lib/x86_64-linux-gnu/libc.so.6 (gsignal+0xcb) [0x7f195c2fb18b]
[    19.755] (EE) 3: /lib/x86_64-linux-gnu/libc.so.6 (abort+0x12b) [0x7f195c2da859]
[    19.756] (EE) unw_get_proc_name failed: no unwind info found [-10]
[    19.756] (EE) 4: /lib/x86_64-linux-gnu/libc.so.6 (?+0x0) [0x7f195c2da71a]
[    19.757] (EE) 5: /lib/x86_64-linux-gnu/libc.so.6 (__assert_fail+0x46) [0x7f195c2ebf36]
[    19.757] (EE) 6: /usr/lib/xorg/Xorg (dixRegisterPrivateKey+0x239) [0x560c4396fa79]
[    19.758] (EE) 7: /usr/lib/xorg/modules/libglamoregl.so (glamor_init+0xcf) [0x7f194c1d735f]
[    19.759] (EE) unw_get_proc_name failed: no unwind info found [-10]
[    19.759] (EE) 8: /usr/lib/xorg/modules/drivers/modesetting_drv.so (?+0x0) [0x7f195bc9d740]
[    19.760] (EE) unw_get_proc_name failed: no unwind info found [-10]
[    19.760] (EE) 9: /usr/lib/xorg/modules/drivers/modesetting_drv.so (?+0x0) [0x7f195bc95b80]
[    19.760] (EE) 10: /usr/lib/xorg/Xorg (AddGPUScreen+0xf5) [0x560c439512d5]
[    19.761] (EE) 11: /usr/lib/xorg/Xorg (xf86PlatformMatchDriver+0xa44) [0x560c439acca4]
[    19.761] (EE) 12: /usr/lib/xorg/Xorg (xf86PlatformDeviceCheckBusID+0x225) [0x560c439b1fe5]
[    19.761] (EE) 13: /usr/lib/xorg/Xorg (config_fini+0xa4a) [0x560c439ae88a]
[    19.762] (EE) 14: /usr/lib/xorg/Xorg (config_fini+0x1408) [0x560c439afcf8]
[    19.762] (EE) 15: /usr/lib/xorg/Xorg (OsCleanup+0x5c1) [0x560c43ab2fa1]
[    19.762] (EE) 16: /usr/lib/xorg/Xorg (WaitForSomething+0x193) [0x560c43aab803]
[    19.763] (EE) 17: /usr/lib/xorg/Xorg (SendErrorToClient+0x117) [0x560c43950cf7]
[    19.763] (EE) 18: /usr/lib/xorg/Xorg (InitFonts+0x3b4) [0x560c43954fc4]
[    19.764] (EE) 19: /lib/x86_64-linux-gnu/libc.so.6 (__libc_start_main+0xf3) [0x7f195c2dc0b3]
[    19.764] (EE) 20: /usr/lib/xorg/Xorg (_start+0x2e) [0x560c4393ea2e]
[    19.765] (EE) 
[    19.765] (EE) 
Fatal server error:
[    19.765] (EE) Caught signal 6 (Aborted). Server aborting
[    19.765] (EE) 
[    19.765] (EE) 
```

But, how  explain that when I install Xubuntu 20.04 on the hard disk of this PC everything works perfectly.

In LTSP mode ,I think the problem is somewhere else, like DBUS or something else.

**Comment by Lichtmacher** [comment]
Thank you for your explanations. I'm really thankful.

I made some tries.

          No improvement means: maybe working, mostly black screen.

I'm using a chrootless server in an virtualbox machine.

sudo apt update
sudo apt full-upgrade
sudo apt autoremove
sudo ltsp image /

No improvement.

I read something about timing problems with lightdm (maybe old). Added some pauses.
sudoedit /etc/lightdm/lightdm.conf.d/10-xubuntu.conf 
[Seat:*]
display-setup-script=/usr/local/bin/sleep.sh

sudoedit /usr/local/bin/sleep.sh
#! /bin/bash
#
sleep 5s
exit 0

sudo chmod 755 /usr/local/bin/sleep.sh 

No improvement.

Direct before starting lightdm:
sudoedit /etc/init/lightdm.conf
added sleep 5s

    sleep 5s
    exec lightdm

No improvement.

Tried another greeter:
sudo apt install lightdm-settings slick-greeter
sudoedit /etc/lightdm/lightdm.conf.d/10-xubuntu.conf
sudo apt install lightdm-settings slick-greeter

sudoedit /etc/lightdm/lightdm.conf.d/10-xubuntu.conf 
[Seat:*]
greeter-session=slick-greeter

No impovement.

 
Installed with synaptic lxdm. Changed from lightdm to lxdm.
cat /etc/X11/default-display-manager
/usr/sbin/lxdm

No improvement.


The strange thing is: sometimes I get a working Login screen. It's seldom, but directly after a reboot of the LTS-server it works always.
Why that?

I would like to troubleshoot by login in to the client. But adding KEEP_SYSTEM_SERVICES="ssh" to /etc/ltsp/ltsp.conf leads to "no hostkey available" and a not working ssh-server on the client.

**Comment by alkisg** [comment]
It sounds like you guys have a Xorg problem and not a greeter problem. E.g. maybe it's related to a specific graphics driver.

I haven't ever seen graphic drivers having any specific issues with LTSP. When they have issues, it affects both LTSP and local installations.

Regarding sshd on the clients, it's a security issue to have the SSH keys publicly readable in NFS or TFTP, so it's disabled by default, and epoptes.org is suggested instead, which uses reverse connections and needs no client-side secrets. Those that really insist can enable sshd by reading https://github.com/ltsp/ltsp/discussions/310.

**Comment by Lichtmacher** [comment]
@alkisg  Thank you, for your help.
I tried to get a bit control of what happens at the start of lightdm and realized, that my changes `/etc/init/lightdm.conf` don't work at all!

The maintainer of the paket lightdm added the startup scripts for initV:
config files in
`/etc/init.d`
and for upstart:
config files in
`/etc/init`
Confusing, because in xubuntu 20.04 systemd is used! What means:
`/lib/systemd/system/lightdm.service` is used for startup.

Making an own configuration file:
```
admini@LTS3:/etc/systemd/system$ sudo cp -p /lib/systemd/system/lightdm.service .
admini@LTS3:/etc/systemd/system$ sudoedit lightdm.service
```
Adding:
`ExecStartPre=/usr/bin/sleep 2s`
And then:
`systemctl daemon-reload`

Now the clients boot.
I think it is a problem of timing. Not lightdm, not X-Server.
Maybe my net, the server and the clients are to fast?!

**Reply by Soylor** [reply]
Sleep worked for me too..

**Comment by FranckGoirand** [comment]
Thank you very much @Lichtmacher  and @alkisg  **it works for me to** !!!! 
It's not a driver problem. Perhaps order of startup services .....

**Comment by Soylor** [comment]
Any way to inlude delay using **ltsp.conf** ?
